packages/lib-classifier/src/components/Classifier/Classifier.js

```diff
@@ -44,12 +44,23 @@ export default function Classifier({
 
   useEffect(function onURLChange() {
     const { workflows } = classifierStore
-    if (workflowID) {
+    if (workflowID && workflowSnapshot) {
       console.log('starting new subject queue', { workflowID, subjectSetID, subjectID })
+      workflowSnapshot.subjectSet = subjectSetID
+      workflows.setResources([workflowSnapshot])
       workflows.selectWorkflow(workflowID, subjectSetID, subjectID, canPreviewWorkflows)
     }
-  }, [canPreviewWorkflows, subjectID, subjectSetID, workflowID])
+  }, [canPreviewWorkflows, subjectID, subjectSetID, workflowID, workflowSnapshot])
 
+  useEffect(function onWorkflowStringsChange() {
+    const { workflows, subjects } = classifierStore
+    if (workflowSnapshot) {
+      // pass the subjectSetID prop into the store as part of the new workflow data
+      workflowSnapshot.subjectSet = subjectSetID
+      console.log('Refreshing workflow strings', workflowSnapshot.id)
+      workflows.setResources([workflowSnapshot])
```

Comment:
Shouldn’t this update workflow strings, not the whole snapshot?

Reply:
Thinking back, I think I did this because the store doesn't update if you only change `workflow.strings`. Something to look into, when there's time. Could be due to `workflow.strings` being frozen.

Reply:
I think I’ll leave a comment in this function. Right now, the tasks only update if the whole workflow snapshot is changed here.